### PR TITLE
Add 'USE_CUDA_GRAPH' Configuration Option

### PR DESCRIPTION
**Pull request by apavliuk-altran**
### Details:
- *Add 'USE_CUDA_GRAPH' Configuration Option for future use with Cuda Graphs*
### Ticket:
- 107838

### CODE REVIEW

File: modules/nvidia_plugin/include/nvidia/nvidia_config.hpp [hunk 1]
```diff
@@ -47,19 +47,22 @@ namespace CUDAConfigParams {
...
- * @brief Defines if optimization should be run for CUDA libraries ("NVIDIA_YES", "NVIDIA_NO" - default).
+ * @brief Defines if benchmarks should be run to determine fastest algorithms for some operations (e.g. Convolution)
+ * ("YES", "NO" - default).
  */
 DECLARE_NVIDIA_CONFIG_KEY(OPERATION_BENCHMARK);
 
+/**
+ * @brief Defines if NVIDIA Plugin should use CUDA graphs for performance acceleration ("YES" - default, "NO").
+ */
+DECLARE_NVIDIA_CONFIG_KEY(USE_CUDA_GRAPH);
```
Comment (nkogteva) [inline modules/nvidia_plugin/include/nvidia/nvidia_config.hpp:65]: General comment: please don't add new options into the legacy properties API

File: modules/nvidia_plugin/README.md [hunk 2]
```diff
@@ -171,7 +171,8 @@ The plugin supports the configuration parameters listed below. All parameters mu
 Parameter name  | Parameter values  | Default  | Description
 ------------- | ------------- | ------------- | -------------
 `NVIDIA_THROUGHPUT_STREAMS`   | `NVIDIA_THROUGHPUT_AUTO`, or non negative integer values  | 1  | Specifies number of CPU "execution" streams for the throughput mode. Upper bound for the number of inference requests that can be executed simultaneously.
-`NVIDIA_OPERATION_BENCHMARK`   | `NVIDIA_YES`, `NVIDIA_NO`  | `NVIDIA_NO`  | Specifies if operation level benchmark should be run for increasing performance of network
+`NVIDIA_OPERATION_BENCHMARK`   | `YES`, `NO`  | `NO`  | Specifies if operation level benchmark should be run for increasing performance of network
+`NVIDIA_USE_CUDA_GRAPH`   | `YES`, `NO`  | `YES`  | Specifies if NVIDIA Plugin should use CUDA graphs for performance acceleration
```
Comment (nkogteva) [inline modules/nvidia_plugin/README.md:175]: Not sure that it good idea to expose new parameter, which turn on nothing. Users are not aware about planning features and it looks very strange.

Moreover, it is questionable whether we will need this parameter in future or we can turn on it always. Can we currently keep this parameter internal for development purposes only?

Comment (apavliuk-altran) [reply modules/nvidia_plugin/README.md]: Removed

Comment (apavliuk-altran) [reply modules/nvidia_plugin/README.md]: Moved `use_cuda_graph` property to `cuda_config.cpp` into `internal` namespace.

**Review by nkogteva — APPROVED** [review]

**Pull request merged**